AMLII-2050 - Add missing logs agent telemetry metrics

ddrthall opened the pull request:
### What does this PR do?
Adds pre-existing telemetry metrics logs.bytes_sent and logs.encoded_bytes to the default agenttelem config. Creates and adds new telemetry metric logs.bytes_missed to record when the log tailer cannot fully consume a file.

### Motivation

### Describe how to test/QA your changes
1. Ensure the correct settings in `datadog.yaml` - log collection and agent telemetry must be enabled, and shrinking the log close timeout greatly speeds up the test time. 
```
agent_telemetry:
  enabled:true
logs_enabled: true
logs_config:
  close_timeout: 1
```

2. Set up a new log file tailer in checks.d and get the tailer running in the agent
```
logs:
  - type: file
    path: "Path/to/tempfile.log"
```

3. Confirm that EncodedBytesSent and BytesSent are being appropriately updated in the exposed expvar statistics : `curl localhost:5000/debug/vars | jq '."logs-agent"'` 

4. Establish a process that regularly generates data for the log file pointed to by the new tailer and let it run for some small amount of time. Then, rotate the log file without pausing the data generation to that file. Do this for two of the three main types of log file rotation: deleting the original file and moving the original file.  
Notice: truncation of the original file, the third common form of log rotation, won't register any lost bytes. This is a limitation of the underlying tailer and was not worth adjusting for this change. 

5. Confirm that BytesMissed is being appropriately updated in the expvar statistics.

6. Confirm the three metrics, datadog.logs.bytes_sent, datadog.logs.encoded_bytes_sent, and datadog.logs.bytes_missed have been successfully exported upstream into the associated datadog org. 

### Possible Drawbacks / Trade-offs

### Additional Notes

gh123man commented on pkg/logs/metrics/metrics.go:
```diff
@@ -58,6 +58,11 @@ var (
+	// BytesDropped is the number of bytes lost before they could be consumed by the agent, such as after a log rotation
+	BytesDropped = expvar.Int{}
+	// TlmBytesDropped is the number of bytes lost before they could be consumed by the agent, such as after log rotation
+	TlmBytesDropped = telemetry.NewCounter("logs", "bytes_dropped",
```
NIT: I wonder if we should call this `bytes_missed` instead of dropped. 
We already have the concept of dropped logs/payloads where the agent explicitly throws them away. In this case the agent doesn't have a chance to tail them. 

(Open to opinions here)

ddrthall replied:
Two thumbs up, I like that better

ddrthall committed:
AMLII-2050 - Add missing logs agent telemetry metrics

Add pre-existing telemetry metrics logs.bytes_sent and logs.encoded_bytes to the default agenttelem config.
Create and add new telemetry metric logs.bytes_missed to record when the log tailer cannot fully consume a file.

ddrthall committed:
AMLII-2050: include release notes

ddrthall committed:
Merge branch 'main' into ryan.hall/add-missing-logs-agent-telemetry-metrics

jhgilbert commented on releasenotes/notes/Add-missing-logs-agent-telemetry-metrics-311c882dd04a0060.yaml:
```diff
@@ -0,0 +1,11 @@
+---
+enhancements:
+  - |
+    Added additional agent telemetry metrics for the log tailer code flow: logs.bytes_sent, logs.encoded_bytes_sent, and logs.bytes_missed
```
```suggestion
    Added additional Agent telemetry metrics for the log tailer code flow: logs.bytes_sent, logs.encoded_bytes_sent, and logs.bytes_missed
```

jhgilbert reviewed (approved):
Approved with a minor suggestion for style, thank you!

ddrthall committed:
Update releasenotes/notes/Add-missing-logs-agent-telemetry-metrics-311c882dd04a0060.yaml

Correct capitalization

Co-authored-by: Jen Gilbert <j.h.gilbert@gmail.com>

ddrthall commented:
/merge
